data/random-teams.js

```diff
@@ -2537,6 +2539,247 @@ class RandomTeams extends Dex.ModdedDex {
+		for (let i = 0; i < setData.set.moves.length; i++) {
+			let moveSlot = setData.set.moves[i];
+			moves.push(setData.moveVariants ? moveSlot[setData.moveVariants[i]] : moveSlot[this.random(moveSlot.length)]);
+		}
+
+		let items = [];
+		if (Array.isArray(setData.set.item) === true) {
+			let randomItem = setData.set.item;
+			items.push(setData.itemVariants ? randomItem[setData.itemVariants] : randomItem[this.random(randomItem.length)]);
+		} else {
+			items.push(setData.set.item);
+		}
+
+		let abilities = [];
+		if (Array.isArray(setData.set.ability) === true) {
+			let randomAbility = setData.set.ability;
+			abilities.push(setData.abilityVariants ? randomAbility[setData.abilityVariants] : randomAbility[this.random(randomAbility.length)]);
+		} else {
+			abilities.push(setData.set.ability);
+		}
+
+		return {
+			name: setData.set.name || template.baseSpecies,
+			species: setData.set.species,
+			gender: setData.set.gender || template.gender || (this.random(2) ? 'M' : 'F'),
+			item: items + '' || setData.set.item || '',
+			ability: abilities + '' || setData.set.ability || template.abilities['0'],
+			shiny: typeof setData.set.shiny === 'undefined' ? !this.random(1024) : setData.set.shiny,
+			level: setData.set.level || 100,
+			happiness: typeof setData.set.happiness === 'undefined' ? 255 : setData.set.happiness,
+			evs: setData.set.evs || {hp: 84, atk: 84, def: 84, spa: 84, spd: 84, spe: 84},
+			ivs: setData.set.ivs || {hp: 31, atk: 31, def: 31, spa: 31, spd: 31, spe: 31},
+			nature: setData.set.nature || 'Serious',
+			moves: moves,
+		};
+	}
+	randomFactoryTeam(depth = 0) {
+		let forceResult = (depth >= 4);
+
+		// The teams generated depend on the tier choice in such a way that
+		// no exploitable information is leaked from rolling the tier in getTeam(p1).
+		let availableTiers = ['Uber', 'OU', 'UU', 'RU', 'NU', 'PU', 'LC', 'Mono'];
+		if (!this.FactoryTier) this.FactoryTier = availableTiers[this.random(availableTiers.length)];
+		const chosenTier = this.FactoryTier;
+
+		let pokemon = [];
+		let pokemonPool = Object.keys(this.randomFactorySets[chosenTier]);
+
+		let typePool = Object.keys(this.data.TypeChart);
+		const type = typePool[this.random(typePool.length)];
+
+		let teamData = {typeCount: {}, typeComboCount: {}, baseFormes: {}, megaCount: 0, zCount: 0, has: {}, forceResult: forceResult, weaknesses: {}, resistances: {}};
+		let requiredMoveFamilies = ['hazardSet', 'hazardClear'];
+		let requiredMoves = {'stealthrock': 'hazardSet', 'rapidspin': 'hazardClear', 'defog': 'hazardClear'};
+		let weatherAbilitiesSet = {'drizzle': 'raindance', 'drought': 'sunnyday', 'snowwarning': 'hail', 'sandstream': 'sandstorm'};
+		let resistanceAbilities = {
+			'dryskin': ['Water'], 'waterabsorb': ['Water'], 'stormdrain': ['Water'],
+			'flashfire': ['Fire'], 'heatproof': ['Fire'],
+			'lightningrod': ['Electric'], 'motordrive': ['Electric'], 'voltabsorb': ['Electric'],
+			'sapsipper': ['Grass'],
+			'thickfat': ['Ice', 'Fire'],
```

Comment:
Is this one intended? The other abilities actually make it so the move won’t do any damage while this one is actually a resistance abilities

Reply:
It was listed alongside these other abilities in the original Gen 6 Battle Factory, so I figured it should probably be here too.